pp/go/storage/loader.go

```diff
@@ -249,7 +279,16 @@ func (l *ShardDataLoader) Load() error {
 		return err
 	}
 
-	return l.createShardWal(shardWalFileName, decoder)
+	if readOnly {
+		return nil
+	}
+
+	encoder, err := decoder.CreateEncoder()
+	if err != nil {
+		return errors.Join(ErrInvalidEncoderVersion, fmt.Errorf("create encoder from decoder: %w", err))
```

Comment:
Why we use `ErrInvalidEncoderVersion`?  Maybe we should use `err`? Because we shouldn't know what kind of error return `decoder.CreateEncoder()`
```suggestion
		return err
```

Reply:
It may be wrap here but not join with anonymous error

Reply:
done